dice_ml/diverse_counterfactuals.py

```diff
@@ -6,132 +6,38 @@
 class CounterfactualExamples:
     """A class to store and visualize the resulting counterfactual explanations."""
 
-    def __init__(self, data_interface, test_instance, test_pred, final_cfs, final_cfs_preds, final_cfs_sparse=None, cfs_preds_sparse=None, posthoc_sparsity_param=0, desired_class="opposite", encoding='one-hot'):
+    def __init__(self, data_interface, final_cfs_df, test_instance_df, final_cfs_df_sparse, posthoc_sparsity_param=0, desired_class="opposite", encoding='one-hot'):
 
         self.data_interface = data_interface
-        self.test_instance = test_instance
-        self.test_pred = test_pred
-        self.final_cfs = final_cfs
-        self.final_cfs_preds = final_cfs_preds
-        self.final_cfs_sparse = final_cfs_sparse
-        self.cfs_preds_sparse = cfs_preds_sparse
-        self.final_cfs_df = None
+        self.final_cfs_df = final_cfs_df
+        self.test_instance_df = test_instance_df
+        self.final_cfs_df_sparse = final_cfs_df_sparse
+
         self.final_cfs_list = None
         self.posthoc_sparsity_param = posthoc_sparsity_param # might be useful for future additions
+        self.test_pred = self.test_instance_df[self.data_interface.outcome_name].iloc[0]
         if desired_class == "opposite":
             self.new_outcome = 1.0 - round(self.test_pred)
         else:
             self.new_outcome = desired_class
         self.encoding = encoding
```

Comment:
removed convert_to_dataframe() methods as this class now takes in CFs in original raw format (as pd.DataFrame)